app/error.go

```diff
@@ -0,0 +1,180 @@
+package app
+
+// ErrorType defines the type of an error
+type ErrorType string
+
+const (
+	// Internal error
+	ErrorInternal ErrorType = "internal"
+	// NotFound error
+	ErrorNotFound ErrorType = "not_found"
+	// Validation error
+	ErrorValidation ErrorType = "validation"
+	// Permission error
+	ErrorPermission ErrorType = "permission"
+	// Unauthorised error
+	ErrorUnauthorised ErrorType = "unauthorised"
+	// Conflict error
+	ErrorConflict ErrorType = "conflict"
+	// Timeout error
+	ErrorTimeout ErrorType = "timeout"
+	// Cancelled error
```

Comment:
You forgot to update the comment

It should be ErrorCancelled

While you did it for severity